sponsors/models.py

```diff
@@ -222,9 +229,20 @@ def __str__(self):
     def _short_name(self):
         return truncatechars(self.name, 42)
 
+    def name_for_display(self, package=None):
+        name = self.name
+        for feature in self.features_config.all():
+            if isinstance(feature, TieredQuantityConfiguration) and feature.package == package:
+                name = feature.display_modifier(name)
```

Comment:
I think that maybe we can refactor this code as well. We could refactor the `BenefitFeatureConfig.display_modifier` to accept  `**kwargs` alongside with the `name` parameter. That way we could move any type of conditional checking to the specific configuration instance. I think we could end up with something like:


```python
def name_for_display(self, package=None):
    name = self.name
    for feature in self.features_config.all():
        name = feature.display_modifier(name, package=package)
    return name
```

And, then, for the `TieredQuantityConfiguration.display_modifier`

```python
def display_modifier(name, **kwargs):
    package = kwargs.get("package")
    if package == self.package:
        return f"{name} ({self.quantity})"
    else:
       return name
```

This will be easier for us to unit test as well.